packages/focusmanager/src/react/use-focus-manager.ts

```diff
@@ -13,35 +14,43 @@ const FOCUSABLE_SELECTORS = [
   'video[controls]'
 ]
 
-const defaultOptions = {
-  autofocus: true,
-  returnFocus: true,
-  trapped: true
-}
-
-export default function useFocusManager(ref, _options) {
-  const opts = { ...defaultOptions, ..._options }
-
-  const prevActiveElRef = React.useRef()
+export default function useFocusManager(
+  ref: React.RefObject<HTMLDivElement>,
+  {
+    autofocus,
+    returnFocus,
+    trapped
+  }: {
+    autofocus: boolean
+    returnFocus: boolean
+    trapped: boolean
+  }
+) {
+  const prevActiveElRef = React.useRef<HTMLElement>(null)
   React.useEffect(() => {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
```

Comment:
Why the ts-ignore here?  I'd like to avoid in all except the most extreme needs.

Reply:
We're overriding a read only property and React types don't allow this but our component's original code does this for functionality.

Reply:
Interesting that the react typings don't allow it but it's one of the "advanced use cases" for refs. Too bad.